packages/deploy/src/stateTransform.ts

```diff
@@ -183,25 +178,23 @@ export function mergeSpecIntoState(
   const nextWorkflows = Object.fromEntries(
     splitZip(oldState.workflows, spec.workflows).map(
       ([workflowKey, stateWorkflow, specWorkflow]) => {
-        stateWorkflow = stateWorkflow || {};
-
         const nextJobs = mergeJobs(
-          stateWorkflow.jobs || {},
+          stateWorkflow?.jobs || {},
```

Comment:
This is such a strange one. Even though `stateWorkflow` will always have a value, typescript insists on saying `stateWorkflow is possibly 'null'`.

I don't see how that can possibly be so, so I've added optional chaining which should workaround it. It's a pretty inelegant fix though.